tests/test_ai_utils.py

```diff
@@ -62,5 +62,25 @@ def test_multiple_sql_code_blocks(self):
         self.assertEqual(AIUtils.extract_code_blocks(text), ["SELECT a FROM b;", "SELECT c FROM d;"])
 
 
+class TestRetryExecution(unittest.TestCase):
```

Comment:
@xinrong-meng can we have a test case which mocks the response  from LLM and generates bad code?